### PR TITLE
feat: add API for showing/hiding toolbar of crud

### PR DESCRIPTION
**Pull request by hdamr**
## Description

This PR adds Java API for letting developers to hide/show toolbar of crud component. it is done by setting an attribute on crud component and the toolbar is hide/shown by changing css display property.

Fixes vaadin/flow-components#1592

## Type of change

- [ ] Bugfix
- [x] Feature

## Checklist

- [x] I have read the contribution guide: https://vaadin.com/docs-beta/latest/guide/contributing/overview/
- [x] I have added a description following the guideline.
- [x] The issue is created in the corresponding repository and I have referenced it.
- [x] I have added tests to ensure my change is effective and works as intended.
- [x] New and existing tests are passing locally with my change.
- [x] I have performed self-review and corrected misspellings.

#### Additional for `Feature` type of change

- [x] Enhancement / new feature was discussed in a corresponding GitHub issue and Acceptance Criteria were created.

### CODE REVIEW

File: vaadin-crud-flow-parent/vaadin-crud-flow-integration-tests/src/test/java/com/vaadin/flow/component/crud/test/BasicUseIT.java [hunk 1]
```diff
@@ -131,6 +131,29 @@ public void crudAndGeneratedGridReactToThemeVariantChanges() {
         Assert.assertNotEquals("no-border", grid.getAttribute("theme"));
     }
 
+    @Test
+    public void hideToolbar() {
+
+        CrudElement crud = $(CrudElement.class).waitForFirst();
+
+        ButtonElement hideToolbarButton = getTestButton("hideToolbarButton");
```
Comment (sissbruecker) [inline vaadin-crud-flow-parent/vaadin-crud-flow-integration-tests/src/test/java/com/vaadin/flow/component/crud/test/BasicUseIT.java:139]: There's no such button

**Review by sissbruecker — COMMENTED** [review]
I'm a bit confused about the state of this PR, it looks like it is not ready?

For one the integration tests don't have a setup / button that would allow toggling the toolbars visibility.

I also can not find any references to this `hide-toolbar` attribute in the web component, so the current solution likely does nothing apart from setting some attribute.

This brings up another issue with the integration tests: Testing that an attribute is present might not be enough, instead we should consider testing the actual visibility of the element.

**Comment by sissbruecker** [comment]
Also noting the acceptance criteria here: https://github.com/vaadin/platform/issues/2311

Especially this point needs some checking:
> When hidden, toolbar must be hidden also from screen readers

File: vaadin-crud-flow-parent/vaadin-crud-flow/src/test/java/com/vaadin/flow/component/crud/CrudTest.java [hunk 2]
```diff
@@ -57,6 +57,17 @@ public void getEditorPosition_defaultOVERLAY() {
...
 
+    @Test
+    public void getCrudToolbarAttribute() {
+        Assert.assertTrue(systemUnderTest.getToolbarVisible());
+    }
+
+    @Test
+    public void hideToolbarAttributePresent() {
+        systemUnderTest.setToolbarVisible(false);
+        Assert.assertEquals(false, systemUnderTest.getToolbarVisible());
```
[the remark this answers is not in the document]
Comment (hdamr) [reply vaadin-crud-flow-parent/vaadin-crud-flow/src/test/java/com/vaadin/flow/component/crud/CrudTest.java:68]: `getAttribute` didn't work on unit tests, I think that has something to do with `flow` or a browser not being present in unit tests, This check has been done here:
https://github.com/vaadin/flow-components/blob/acfd50c328f891dfb3833c62dc2b79a9486a3d9a/vaadin-crud-flow-parent/vaadin-crud-flow-integration-tests/src/test/java/com/vaadin/flow/component/crud/test/BasicUseIT.java#L142

**Comment by hdamr** [comment]
As far as I know, setting display property of an element to `none` will remove it from DOM and hide it from both screen readers and browser.
https://github.com/vaadin/web-components/blob/e942485891560bbaceef66b008f413047648df49/packages/vaadin-crud/src/vaadin-crud.js#L181

File: vaadin-crud-flow-parent/vaadin-crud-flow-integration-tests/src/test/java/com/vaadin/flow/component/crud/test/BasicUseIT.java (hunk 1 above)
Comment (hdamr) [reply vaadin-crud-flow-parent/vaadin-crud-flow-integration-tests/src/test/java/com/vaadin/flow/component/crud/test/BasicUseIT.java:139]: I accidentally had reverted this file.

**Comment by sissbruecker** [comment]
> As far as I know, setting display property of an element to `none` will remove it from DOM and hide it from both screen readers and browser.
> https://github.com/vaadin/web-components/blob/e942485891560bbaceef66b008f413047648df49/packages/vaadin-crud/src/vaadin-crud.js#L181

That's not correct, an element will only be removed from the DOM when you call `remove()`. Setting a CSS property can not affect the DOM structure.

**Comment by hdamr** [comment]
> > As far as I know, setting display property of an element to `none` will remove it from DOM and hide it from both screen readers and browser.
> > https://github.com/vaadin/web-components/blob/e942485891560bbaceef66b008f413047648df49/packages/vaadin-crud/src/vaadin-crud.js#L181
> 
> That's not correct, an element will only be removed from the DOM when you call `remove()`. Setting a CSS property can not affect the DOM structure.

I am referring to this [link](https://css-tricks.com/places-its-tempting-to-use-display-none-but-dont/)

**Comment by sissbruecker** [comment]
> I am referring to this link

I missed that you referred to my comment about accessibility, it would be helpful if you quote which part of the review you want to address. Still, the statement is not correct, the element will not be removed from the DOM.

What I would like to see addressed is this point:
> This brings up another issue with the integration tests: Testing that an attribute is present might not be enough, instead we should consider testing the actual visibility of the element.

I think we should change the ITs to actually test the visibility of the toolbar, rather than just checking for attributes. It's an integration test after all.

**Comment by web-padawan** [comment]
@hdamr the web component version is now updated so the validation build for this PR passed.

**Review by sissbruecker — APPROVED** [review]

**Pull request merged**